Source/WTF/wtf/text/StringBuilder.h

```diff
@@ -87,9 +86,9 @@ class StringBuilder {
     UChar operator[](unsigned i) const;
 
     bool is8Bit() const;
-    std::span<const LChar> span8() const { return { characters<LChar>(), length() }; }
-    std::span<const UChar> span16() const { return { characters<UChar>(), length() }; }
-    template<typename CharacterType> std::span<const CharacterType> span() const { return std::span(characters<CharacterType>(), length()); }
+    std::span<const LChar> span8() const { return unsafeMakeSpan(characters<LChar>(), length()); }
+    std::span<const UChar> span16() const { return unsafeMakeSpan(characters<UChar>(), length()); }
+    template<typename CharacterType> std::span<const CharacterType> span() const { return unsafeMakeSpan(characters<CharacterType>(), length()); }
```

Comment:
Kind of sad that we have a span-compatible type internally (`m_string` / `m_buffer`), but `characters<T>()` converts it to raw pointer, and then we unsafely convert back to span here.

I wonder if we can invert the dependency. Move the logic from `characters<T>()` into `span<T>()`, and then have `characters<T>()` wrap `span<T>()` and return `.data()`. That way, we can get rid of the `unsafeMakeSpan()` calls.

Reply:
Ok, I got rid of `characters<T>()` altogether.